packages/sanity/src/core/form/inputs/files/common/FileInputButton/FileInputButton.tsx

```diff
@@ -2,7 +2,7 @@ import React, {useId} from 'react'
 import {ButtonProps} from '../../../../../../ui'
 import {FileButton} from './styles'
 
-export interface FileInputButtonProps extends ButtonProps {
+export type FileInputButtonProps = ButtonProps & {
```

Comment:
Possibly an unnecessary change – please [prefer interfaces over types](https://www.notion.so/sanityio/Studio-Development-Guidelines-f48f9a9fe62843018ccd16c043c238cf?pvs=4#0240ed3590d8483ab8fabc1081ac5e97)!

Reply:
We are going to need it in this case, with the change of the button props to be an intersection it can be extended, because it won't know which of the two intersections to extend.
```
An interface can only extend an object type or intersection of object types with statically known members.ts(2312)
```